torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -26,6 +26,92 @@ class AllGatherResult(NamedTuple):
     all_gather_input_split_sizes: List[int]
 
 
+lib = torch.library.Library("fsdp", "FRAGMENT")  # noqa: TOR901
+
+lib.define(
+    """
+    all_gather_copy_in(
+        SymInt all_gather_input_numel,
+        SymInt world_size,
+        SymInt rank,
+        ScalarType dtype,
+        Device device,
+        SymInt[] inp_split_sizes,
+        Tensor[] all_gather_inputs
+    ) -> (Tensor, Tensor)
+    """
+)
+
+
+@torch.library.impl(lib, "all_gather_copy_in", "Meta")
+def all_gather_copy_in_meta(
+    all_gather_input_numel: int,
+    world_size: int,
+    rank: int,
+    dtype: torch.dtype,
+    device: torch.device,
+    inp_split_sizes: List[int],
+    all_gather_inputs: List[torch.Tensor],
```

Comment:
can we make the list tensor inputs in the first argument instead of the last? This is to make sure operator registration happen in a more consistent manner similar to the core aten operators so that other subsystems could interact with the op better. I would prefer a order roughly like this:

```
(
all_gather_inputs: List[torch.Tensor],
inp_split_sizes: List[int],
all_gather_input_numel: int,
...
)
```